src/main/java/racingcar/domain/Cars.java

```diff
@@ -0,0 +1,79 @@
+package racingcar.domain;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static racingcar.util.RandomUtil.generateRandomNumber;
+
+
+public class Cars {
+
+    private static final String DUPLICATED = "[ERROR] 중복된 자동차 이름은 허용되지 않습니다.";
+
+    private Set<Car> cars = new HashSet<>();
+
+    public Cars() {
+    }
+
+    public Cars(String[] names) {
+        validateDuplication(names);
+
+        for (String name : names) {
+            cars.add(new Car(name));
+        }
+    }
+
+    public Cars(Map<String, Integer> carInfo) {
+        Set<String> names = carInfo.keySet();
+        validateDuplication(names.toArray(new String[names.size()]));
+
+        for (String name : names) {
+            cars.add(new Car(name, carInfo.get(name)));
+        }
+    }
```

Comment:
리뷰를 잘못드렸던 것 같습니다 🙇‍♂️ 

주 생성자를 두고 정적 팩토리 메서드를 사용하면 좋을 것 같습니다!

* [정적 팩토리 메서드(Static Factory Method)는 왜 사용할까?](https://tecoble.techcourse.co.kr/post/2020-05-26-static-factory-method/)

```suggestion    
    public Cars(Set<Car> cars) {
        validateDuplication(cars);
        this.cars = cars;
    }
    
    public static Cars of(String[] names) {
        Set<Car> cars = new HashSet<>();
        for (String name : names) {
            cars.add(new Car(name));
        }
        
        return new Cars(cars);
    }
```


다만 아래 생성자는 오직 테스트를 위해 생성된 생성자로 보입니다.

```java
public Cars(Map<String, Integer> carInfo) {
    Set<String> names = carInfo.keySet();
    validateDuplication(names.toArray(new String[names.size()]));

    for (String name : names) {
        cars.add(new Car(name, carInfo.get(name)));
    }
}
```

Reply:
정적 팩토리 메소드.. 핀 덕분에 처음 알게되었습니다! 덕분에 다음 리뷰 요청이 늦어졌네요..ㅎㅎㅎㅎㅎ 덕분에 테스팅 하기가 훨씬 수월한 코드가 완성되었습니다! 다만 공부하다보니 of는 여러개의 매개변수를 받아서 객체를 생성하는 메서드명, from은 한개의 매개변수를 받아서 객체를 생성하는 메서드명으로 선정하는 것이 컨벤션이라고 하여 from으로 선언했습니다!